src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -371,15 +371,38 @@ private async Task ProcessJob()
                     }
                 }
 
-                // if our received CancellationToken is cancelled we should
+                    // Get the latest version of the reindex job in case another thread has updated it
+                    // for most cases if another process updates the job (such as a DELETE request)
+                    // the _etag change will cause a JobConflict exception and this task will be aborted
+                    // but here we add one more check before attempting to mark the job as complete,
+                    // or starting another iteration of the loop
+                    await jobSemaphore.WaitAsync();
+                    try
+                    {
+                    using (IScoped<IFhirOperationDataStore> store = _fhirOperationDataStoreFactory.Invoke())
+                        {
+                            var wrapper = await store.Value.GetReindexJobByIdAsync(_reindexJobRecord.Id, cancellationToken);
+                            _weakETag = wrapper.ETag;
+                            _reindexJobRecord = wrapper.JobRecord;
+                        }
+                    }
+                    finally
+                    {
+                        jobSemaphore.Release();
```

Comment:
>jobSemaphore [](http://example.com/codeflow?start=24&length=12)

should be _jobSemaphore and _cancellationToken. Sorry for messing with that code